module_3/gift_certificates_system/service/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.OrderDao;
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.dto.OrderDto;
+import com.epam.esm.dto.UserDto;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.Order;
+import com.epam.esm.entity.User;
+import com.epam.esm.exception.EntityNonExistentException;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.util.EsmPagination;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.epam.esm.util.MessagePropertyKey.EXCEPTION_GIFT_CERTIFICATE_ID_NOT_FOUND;
+import static com.epam.esm.util.MessagePropertyKey.EXCEPTION_ORDER_FOR_USER_NOT_FOUND;
+import static com.epam.esm.util.MessagePropertyKey.EXCEPTION_ORDER_ID_NOT_FOUND;
+import static com.epam.esm.util.MessagePropertyKey.EXCEPTION_UNSUPPORTED_OPERATION;
+import static com.epam.esm.util.MessagePropertyKey.EXCEPTION_USER_ID_NOT_FOUND;
+
+/**
+ * Order service implementation.
+ */
+@Service
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private final ModelMapper modelMapper;
+    private final GiftCertificateServiceImpl gcService;
+    private final OrderDao orderDao;
+    private final UserDao userDao;
+    private final GiftCertificateDao gcDao;
+
+    /**
+     * Instantiates a new tag service.
+     *
+     * @param modelMapper Model mapper.
+     * @param orderDao    Tag DAO layer.
+     * @param userDao     User DAO layer.
+     * @param gcDao       Gift certificate DAO layer.
+     * @param gcService   Gift certificate service layer.
+     */
+    public OrderServiceImpl(ModelMapper modelMapper, GiftCertificateServiceImpl gcService, OrderDao orderDao, UserDao userDao, GiftCertificateDao gcDao) {
+        this.modelMapper = modelMapper;
+        this.gcService = gcService;
+        this.orderDao = orderDao;
+        this.userDao = userDao;
+        this.gcDao = gcDao;
```

Comment:
Please use meaningful variable names instead of gcService and gcDao, the readability of your code is much more important then "compactness" of it

Reply:
Thanks, renamed to `certificateService` and `certificateDao` respectively.